### DISCUSSION TITLE
Background image interfering with overlay image

### DISCUSSION THREAD

**Discussion by MorpheuTK** [ImageMagick/ImageMagick#2217] [Help] [answered]
I'm making a Twitter bot that takes two random images and merges them using imagemagick, and I can't figure it out why sometimes the background image interferies with the overlay image, I tested with other backgrounds and it appears to be only the black and white ones that affect the overlay image. 
[Here's the image](https://imgur.com/a/n0TWTn2), the girl in front had colors, but after using imagemagick she became black and white. I did this manually just to make sure it wasn't my code that had something wrong, but the result was the same. Thank you in advance.

My command line: magick convert "flags/Anarcho-Pacifist.png" "girls/Izumi Konata.png" -gravity center geometry +200+0 -composite output/result.png

**Comment by fmw42** [comment]
You are likely overlaying a color image on to a grayscale image. The output will be the same as the first image in the command line if you do nothing else. You can force to 24-bit color by prefacing your output with PNG24: as follows. Note for IM 7, use magick, not magick convert.

`magick "flags/Anarcho-Pacifist.png" "girls/Izumi Konata.png" -gravity center geometry +200+0 -composite PNG24:output/result.png`

**Reply by MorpheuTK** [reply]
Hmm no, the result was the same, all black and white.

**Reply by fmw42** [reply]
Can you provide both images so we can test with them. I only see one image linked in your post.

**Reply by MorpheuTK** [reply]
> Insert "-colorspace sRGB" after reading the first file.
> 
> If that doesn't work, please link to sample inputs.

This worked, thank you for your help. :)

**Comment by snibgo** [comment, marked as answer]
Insert "-colorspace sRGB" after reading the first file.

If that doesn't work, please link to sample inputs.

**Reply by MorpheuTK** [reply]
It worked! Thank you very much.
